rs/types/types/src/lib.rs

```diff
@@ -522,37 +522,59 @@ impl TryFrom<NumBytes> for MemoryAllocation {
     }
 }
 
-/// Allow an object to report its own byte size. It is only meant to be an
-/// estimate, and not an exact measure of its heap usage or length of serialized
-/// bytes.
+/// Allow an object to report its own byte size in memory and on disk. It is
+/// only meant to be an estimate, and not an exact measure of its heap usage,
+/// disk usage, or length of serialized bytes.
 pub trait CountBytes {
-    fn count_bytes(&self) -> usize;
+    fn memory_count_bytes(&self) -> usize;
+    fn disk_count_bytes(&self) -> usize {
+        0
+    }
 }
```

Comment:
Just curious: what's the benefit of extending the `CountBytes` trait vs. defining a new trait to be implemented by the types relevant to the LRU cache? Are the latter a large proportion of the types that already implement `CountBytes`?

Reply:
Maybe it actually is easier to make a new trait. I'll try it out.